pkg/reporter/reporter.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package reporter provides an SDK for reporting Guardian results.
+package reporter
+
+import (
+	"context"
+)
+
+type (
+	// Operation is the operation Guardian is performing.
+	Operation int
+	// Status is the result of the operation Guardian is performing.
+	Status int
+)
+
+// the operations supported by reporters.
+const (
+	PlanOperation Operation = iota
+	ApplyOperation
+	UnknownOperation
+)
+
+// the supported statuses for reporters.
+const (
+	StartStatus Status = iota
+	SuccessStatus
+	FailureStatus
+	NoChangesStatus
+	UnknownStatus
+)
+
+// Params are the parameters for writing statuses.
+type Params struct {
+	Operation     Operation
+	Status        Status
+	IsDestroy     bool
```

Comment:
Why wouldn't this just be `OperationDestroy`?

Reply:
I could add PlanDestroy and ApplyDestroy as additional operations?

Reply:
I don't have a good enough insight to know which one will be better/worse, but I do think like bitbanging could be helpful

`OperationPlan & OperationDestroy` gives you a number that can easily check for "is plan" and "is destroy". You'd need to use bitshifts instead of ints directly (`iota << 1`)